eng/PublishSourceBuild.props

```diff
@@ -81,7 +81,7 @@
       <SourceBuiltLayoutDir>$([MSBuild]::NormalizeDirectory('$(BaseIntermediateOutputPath)', 'artifacts-layout'))</SourceBuiltLayoutDir>
 
       <!-- Outputs -->
-      <SourceBuiltTarballName>$(SourceBuiltAssetsDir)/$(SourceBuiltArtifactsTarballName).$(SourceBuiltSdkNonStableVersion).$(TargetRid)$(ArchiveExtension)</SourceBuiltTarballName>
+      <SourceBuiltTarballName>$(SourceBuiltAssetsDir)/source-build/$(SourceBuiltArtifactsTarballName).$(SourceBuiltSdkNonStableVersion).$(TargetRid)$(ArchiveExtension)</SourceBuiltTarballName>
```

Comment:
I suspect that this extra `source-build` dir may not be needed since it's not part of the artifacts dir that ends up getting published.

Reply:
`$(ArtifactsAssetDir)`: `$(RepoRoot)/artifacts/assets/$(Configuration)`
`$(SourceBuiltAssetsDir)`: `$(Build.ArtifactStagingDirectory)/artifacts/assets/$(Configuration)`
The content under the `$(SourceBuiltAssetsDir) path will be published to Azure Artifacts, because: https://github.com/dotnet/dotnet/blob/15307b3e362d30e1e6cfac69becb28160534db63/eng/pipelines/templates/jobs/vmr-build.yml#L336-L341

So the extra `source-build` dir is necessary, otherwise, we need to change the target publish path.

Reply:
And there is another similar issue, two copies of`dotnet-sdk-*` files are published:
<img width="354" height="317" alt="image" src="https://github.com/user-attachments/assets/5ed8307e-77ab-4a3a-b0dd-8937c3adbb9b" />

Caused by: https://github.com/dotnet/dotnet/blob/main/eng/Publishing.props#L116